server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -3899,6 +3914,15 @@ private UserVm createVirtualMachine(DataCenter zone, ServiceOffering serviceOffe
         return vm;
     }
 
+    private Boolean checkIfDynamicScalingCanBeEnabled(Boolean dynamicScalingEnabled, ServiceOfferingVO offering, VMTemplateVO template, Long zoneId) {
```

Comment:
can we not define this in UserVMManager and call this method at all other places where the check is repeated - eg. at KubernetesClusterStartWorker and other classes